### PR TITLE
8312491: Update Classfile API snippets and examples

### PR DESCRIPTION
**Pull request by asotona**
This pull request updates Classfile API snippets and examples and adds missing javadoc.

Please review.

Thanks,
Adam

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8312491](https://bugs.openjdk.org/browse/JDK-8312491): Update Classfile API snippets and examples (**Sub-task** - P4)

### Reviewers
 * [Jan Lahoda](https://openjdk.org/census#jlahoda) (@lahodaj - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14968/head:pull/14968` \
`$ git checkout pull/14968`

Update a local copy of the PR: \
`$ git checkout pull/14968` \
`$ git pull https://git.openjdk.org/jdk.git pull/14968/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14968`

View PR using the GUI difftool: \
`$ git pr show -t 14968`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14968.diff">https://git.openjdk.org/jdk/pull/14968.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14968#issuecomment-1653727203)

### CODE REVIEW

File: src/java.base/share/classes/jdk/internal/classfile/CodeBuilder.java [hunk 1]
```diff
@@ -1233,6 +1233,10 @@ default CodeBuilder lconst_1() {
         return with(ConstantInstruction.ofIntrinsic(Opcode.LCONST_1));
     }
 
+    default CodeBuilder ldc(ConstantDesc value) {
+        return ldc(BytecodeHelpers.constantEntry(constantPool(), value));
+    }
+
```
[the remark this answers is not in the document]
Comment (asotona) [reply src/java.base/share/classes/jdk/internal/classfile/CodeBuilder.java:2172]: I did first round of filling missing javadoc.

**Comment by asotona** [comment]
The javadoc and snippets are available online at:
https://cr.openjdk.org/~asotona/8312491/classfile-api/javadoc/java.base/jdk/internal/classfile/package-summary.html

**Review by lahodaj — APPROVED** [review]
Looks like an improvement to me!

**Pull request closed without merging**